packages/runtime-core/src/apiWatch.ts

```diff
@@ -285,7 +285,13 @@ export function instanceWatch(
 }
 
 function traverse(value: unknown, seen: Set<unknown> = new Set()) {
-  if (!isObject(value) || seen.has(value)) {
+  if (!isObject(value)) {
+    if (seen.size === 0) {
+      return value
+    }
+    return
+  }
```

Comment:
I think just changing it to `return value` is fine.

```suggestion
  if (!isObject(value) || seen.has(value)) {
    return value
  }
```